geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/DeltaWriter.scala

```diff
@@ -503,17 +494,13 @@ object DeltaWriter extends StrictLogging {
         var resultIndex = 0
         // copy the next sorted value and then queue and sort the next element out of the batch we copied from
         do {
-          val (_, batch, i) = queue.dequeue()
-          val (vector, sort, transfers, mappings) = toMerge(batch)
-          transfers.foreach(_.apply(i, resultIndex))
+          val next = queue.remove()
+          if (next.transfer(resultIndex)) {
+            queue.add(next)
+          }
           result.underlying.setIndexDefined(resultIndex)
           resultIndex += 1
-          val nextBatchIndex = i + 1
```

Comment:
@elahrvivaz can you briefly describe what's happening here with the change in logic?

Reply:
sure - this may not be 100% accurate as I'm going from memory. previously there was an array of attribute readers, one for each batch. the queue contained `(current value, batch number, current reader index)`. We'd pop the top value off the queue, then push the next value read from the array (based on the batch number), and the next reader index. instead, i've moved the state into sortable classes, which will hold the reader, the current reader value, and the reader index, and sort based on the current reader value. we pop one off the queue, then we 'increment' it (read the next value from that batch and increment internal counters), then put it back on the queue with its new value. so we create a lot less classes, and we don't have to track the relationship between the queue values and the batch readers b/c they're the same thing now.